src/textual/_text_area_theme.py

```diff
@@ -0,0 +1,353 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+
+from rich.style import Style
+
+from textual.app import DEFAULT_COLORS
+from textual.color import Color
+from textual.design import DEFAULT_DARK_SURFACE
+
+
+@dataclass
+class TextAreaTheme:
+    """A theme for the `TextArea` widget.
+
+    Allows theming the general widget (gutter, selections, cursor, and so on) and
+    mapping of tree-sitter tokens to Rich styles.
+
+    For example, consider the following snippet from the `markdown.scm` highlight
+    query file. We've assigned the `heading_content` token type to the name `heading`.
+
+    ```
+    (heading_content) @heading
+    ```
+
+    Now, we can map this `heading` name to a Rich style, and it will be styled as
+    such in the `TextArea`, assuming a parser which returns a `heading_content`
+    node is used (as will be the case when language="markdown").
+
+    ```
+    TextAreaTheme('my_theme', syntax_styles={'heading': Style(color='cyan', bold=True)})
+    ```
+
+    We can register this theme with our `TextArea` using the `register_theme` method,
+    and headings in our markdown files will be styled bold cyan.
+    """
+
+    name: str
+    """The name of the theme."""
+
+    base_style: Style | None = None
+    """The background style of the text area. If `None` the parent style will be used."""
+
+    gutter_style: Style | None = None
+    """The style of the gutter. If `None`, a legible TextAreaStyle will be generated."""
+
+    cursor_style: Style | None = None
+    """The style of the cursor. If `None`, the legible TextAreaStyle will be generated."""
```

Comment:
Above it says "a legible", here it says "the legible", the latter sounds like it's referring to something I might find somewhere else. Just a typo or should I read more into it?